.github/workflows/flowglad-next-sharded-unit-tests.yml

```
@@ -0,0 +1,142 @@
+# This workflow runs the unit tests 
+# for flowglad-next in a sharded manner.
+# It runs the pnpm test command across 4 shards,
+# and then merges the reports together.
+name: Flowglad Next Unit Tests
+
+# This workflow is designed to be called by other workflows
+# rather than triggered directly by events
+on:
+  workflow_call:
+
+jobs:
+  # Main job that runs tests across multiple shards in parallel
+  # Each shard runs a subset of tests to improve performance
+  flowglad-next-sharded-tests:
+    runs-on: ubuntu-latest
+    # Matrix strategy creates 4 parallel jobs (shards)
+    # Each shard gets a different shardIndex (1-4) but same shardTotal (4)
+    strategy:
+      matrix:
+        shardIndex: [1, 2, 3, 4]  # Individual shard identifier
+        shardTotal: [4]           # Total number of shards
+      fail-fast: false           # Don't stop other shards if one fails
```

To align with previous lines' comments:

```suggestion
      fail-fast: false            # Don't stop other shards if one fails
```